samcli/commands/local/lib/swagger/parser.py

```diff
@@ -36,7 +45,106 @@ def get_binary_media_types(self):
         """
         return self.swagger.get(self._BINARY_MEDIA_TYPES_EXTENSION_KEY) or []
 
-    def get_routes(self, event_type=Route.API):
+    def get_authorizers(self) -> Dict[str, Authorizer]:
+        """
+        Parse Swagger document and returns a list of Authorizer objects
+
+        Returns
+        -------
+        dict[str, Authorizer]
+            A map of authorizer names and Authorizer objects found in the body definition
+        """
+        authorizers: Dict[str, Authorizer] = {}
+
+        authorizer_dict = {}
+        document_version = self.swagger.get("swagger") or self.swagger.get("openapi") or ""
+
+        if document_version.startswith("2."):
+            LOG.debug("Parsing Swagger document using 2.0 specification")
+            authorizer_dict = self.swagger.get("securityDefinitions", {})
+        elif document_version.startswith("3."):
+            LOG.debug("Parsing Swagger document using 3.0 specification")
+            authorizer_dict = self.swagger.get("components", {}).get("securitySchemes", {})
+        else:
+            raise InvalidOasVersion(
+                f"An invalid OpenApi version was detected: '{document_version}', must be one of 2.x or 3.x",
+            )
+
+        for auth_name, properties in authorizer_dict.items():
+            authorizer_object = properties.get(self._AUTHORIZER_KEY)
+
+            if authorizer_object:
+                valid_types = ["token", "request"]
+
+                authorizer_type = authorizer_object.get("type", "").lower()
+                payload_version = authorizer_object.get("authorizerPayloadFormatVersion", "1.0")
+
+                lambda_name = LambdaUri.get_function_name(authorizer_object.get("authorizerUri"))
+
+                if not lambda_name:
+                    LOG.warning(
+                        "Unable to parse authorizerUri '%s' for authorizer '%s', skipping", lambda_name, auth_name
+                    )
+
+                    continue
+
+                # only add authorizer if it is token or request based (not jwt)
+                if authorizer_type in valid_types and lambda_name:
+                    lambda_authorizer = LambdaAuthorizer(
+                        authorizer_name=auth_name,
+                        type=authorizer_type,
+                        payload_version=payload_version,
+                        lambda_name=lambda_name,
+                        identity_sources=[authorizer_object.get("identitySource")],
```

Comment:
There is a difference in identity source definition locations between the V1 and V2, need to rework this to consider that